File: z-clients/javascript/src/apis/adminAuthPolicy.ts
// this file is @generated

import {
    type AdminAccessPolicyConfigureIn,
    AdminAccessPolicyConfigureInSerializer,
} from '../models/adminAccessPolicyConfigureIn';
import {
    type AdminAccessPolicyConfigureOut,
    AdminAccessPolicyConfigureOutSerializer,
} from '../models/adminAccessPolicyConfigureOut';
import {
    type AdminAccessPolicyDeleteIn,
    AdminAccessPolicyDeleteInSerializer,
} from '../models/adminAccessPolicyDeleteIn';
import {
    type AdminAccessPolicyDeleteOut,
    AdminAccessPolicyDeleteOutSerializer,
} from '../models/adminAccessPolicyDeleteOut';
import {
    type AdminAccessPolicyGetIn,
    AdminAccessPolicyGetInSerializer,
} from '../models/adminAccessPolicyGetIn';
import {
    type AdminAccessPolicyListIn,
    AdminAccessPolicyListInSerializer,
} from '../models/adminAccessPolicyListIn';
import {
    type AdminAccessPolicyOut,
    AdminAccessPolicyOutSerializer,
} from '../models/adminAccessPolicyOut';
import {
    type ListResponseAdminAccessPolicyOut,
    ListResponseAdminAccessPolicyOutSerializer,
} from '../models/listResponseAdminAccessPolicyOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class AdminAuthPolicy {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Create or update an access policy */
    public configure(
        adminAccessPolicyConfigureIn: AdminAccessPolicyConfigureIn,
    ): Promise<AdminAccessPolicyConfigureOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-policy.configure");

        request.setBody(
            AdminAccessPolicyConfigureInSerializer._toJsonObject(adminAccessPolicyConfigureIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAccessPolicyConfigureOutSerializer._fromJsonObject,
        );
    }/** Delete an access policy */
    public delete(
        adminAccessPolicyDeleteIn: AdminAccessPolicyDeleteIn,
    ): Promise<AdminAccessPolicyDeleteOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-policy.delete");

        request.setBody(
            AdminAccessPolicyDeleteInSerializer._toJsonObject(adminAccessPolicyDeleteIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAccessPolicyDeleteOutSerializer._fromJsonObject,
        );
    }/** Get an access policy by ID */
    public get(
        adminAccessPolicyGetIn: AdminAccessPolicyGetIn,
    ): Promise<AdminAccessPolicyOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-policy.get");

        request.setBody(
            AdminAccessPolicyGetInSerializer._toJsonObject(adminAccessPolicyGetIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAccessPolicyOutSerializer._fromJsonObject,
        );
    }/** List all access policies */
    public list(
        adminAccessPolicyListIn: AdminAccessPolicyListIn,
    ): Promise<ListResponseAdminAccessPolicyOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-policy.list");

        request.setBody(
            AdminAccessPolicyListInSerializer._toJsonObject(adminAccessPolicyListIn)
        );
        
        return request.send(
            this.requestCtx,
            ListResponseAdminAccessPolicyOutSerializer._fromJsonObject,
        );
    }
}

File: z-clients/java/src/main/java/com/svix/diom/apis/AdminAuthPolicy.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.AdminAccessPolicyConfigureIn;
import com.svix.diom.models.AdminAccessPolicyConfigureOut;
import com.svix.diom.models.AdminAccessPolicyDeleteIn;
import com.svix.diom.models.AdminAccessPolicyDeleteOut;
import com.svix.diom.models.AdminAccessPolicyGetIn;
import com.svix.diom.models.AdminAccessPolicyListIn;
import com.svix.diom.models.AdminAccessPolicyOut;
import com.svix.diom.models.ListResponseAdminAccessPolicyOut;

public class AdminAuthPolicy {
    private final HttpClient client;

    public AdminAuthPolicy(HttpClient client) {
        this.client = client;
    }

    /** Create or update an access policy */
    public AdminAccessPolicyConfigureOut configure(
        final AdminAccessPolicyConfigureIn adminAccessPolicyConfigureIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-policy.configure",
            null,
            adminAccessPolicyConfigureIn,
            AdminAccessPolicyConfigureOut.class
        );
    }

    /** Delete an access policy */
    public AdminAccessPolicyDeleteOut delete(
        final AdminAccessPolicyDeleteIn adminAccessPolicyDeleteIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-policy.delete",
            null,
            adminAccessPolicyDeleteIn,
            AdminAccessPolicyDeleteOut.class
        );
    }

    /** Get an access policy by ID */
    public AdminAccessPolicyOut get(
        final AdminAccessPolicyGetIn adminAccessPolicyGetIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-policy.get",
            null,
            adminAccessPolicyGetIn,
            AdminAccessPolicyOut.class
        );
    }

    /** List all access policies */
    public ListResponseAdminAccessPolicyOut list(
        final AdminAccessPolicyListIn adminAccessPolicyListIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-policy.list",
            null,
            adminAccessPolicyListIn,
            ListResponseAdminAccessPolicyOut.class
        );
    }

    /** List all access policies */
    public ListResponseAdminAccessPolicyOut list(
        
    ) throws DiomException {
        return this.list(
            new AdminAccessPolicyListIn()
        );
    }
}